### PR TITLE
Revise SDL-0296 - Possibility to update video streaming capabilities during ignition cycle

### PR DESCRIPTION
**Pull request by dboltovskyi**
The main purpose of this proposal is to make it possible to update video streaming capabilities during the ignition cycle.

This is a new revision for initial version [0296-Update-video-streaming-capabilities-during-ignition-cycle](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md) which is mostly address the comments provided in https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-658826750.

It includes the following changes:
- Updated Public API for iOS and Android:
  - Renamed properties and interfaces to `VideoStreamingRange`
  - ~~Aligned iOS and Android properties~~ 
- Updated parameters for `OnAppCapabilityUpdated` notification:
  - removed `AppVideoStreamingCapability` struct
  - renamed `min/max` attributes for `additionalVideoStreamingCapabilities`
- Updated HMI Integration Guidelines
- Updated sequence diagrams:
  - added information on Stop/Start displaying video stream on HMI
- Added HMI as an impacted platform

### CODE REVIEW

**Comment by dboltovskyi** [comment]
@theresalech Please notice new revision is ready for PM review.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 1]
```diff
@@ -125,10 +118,10 @@ While `diagonalScreenSize` is mandatory, `aspectRatio` and `resolution` are opti
 In `SDLStreamingMediaConfiguration`:
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:118]: The following still needs to be added to the iOS code examples: https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-642797352, with `streamingVideoDelegate` changed to `delegate`.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 2]
```diff
@@ -125,10 +118,10 @@ While `diagonalScreenSize` is mandatory, `aspectRatio` and `resolution` are opti
...
 
 ```objectivec
-@property (strong, nonatomic, nullable) SDLSupportedStreamingRange *supportedLandscapeStreamingRange;
-@property (strong, nonatomic, nullable) SDLSupportedStreamingRange *supportedPortraitStreamingRange;
+@property (strong, nonatomic, nullable) VideoStreamingRange *supportedLandscapeStreamingRange;
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:121]:
```suggestion
@property (strong, nonatomic, nullable) SDLVideoStreamingRange *supportedLandscapeStreamingRange;
```

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 3]
```diff
@@ -125,10 +118,10 @@ While `diagonalScreenSize` is mandatory, `aspectRatio` and `resolution` are opti
...
+@property (strong, nonatomic, nullable) VideoStreamingRange *supportedPortraitStreamingRange;
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:122]:
```suggestion
@property (strong, nonatomic, nullable) SDLVideoStreamingRange *supportedPortraitStreamingRange;
```

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 4]
```diff
@@ -125,10 +118,10 @@ While `diagonalScreenSize` is mandatory, `aspectRatio` and `resolution` are opti
...
 
-@interface SDLSupportedStreamingRange : NSObject
+@interface VideoStreamingRange : NSObject
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:124]:
```suggestion
@interface SDLVideoStreamingRange : NSObject
```

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 5]
```diff
@@ -153,30 +146,32 @@ In `VideoStreamManager` add new `startRemoteDisplayStream`:
...
-    private Resolution minSupportedResolution;
-    private Resolution maxSupportedResolution;
-    private Double maxScreenDiagonal;
-    private AspectRatio aspectRatio;
+public class VideoStreamingRange {
+    private Resolution minResolution;
+    private Resolution maxResolution;
+    private Double minAspectRatio;
+    private Double maxAspectRatio;
+    private Double minScreenDiagonal;
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:161]: These changes were not requested from the Steering Committee: https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-658826750.  We understand that they're incorrect, but will need a revision proposal to change, as the changes have not undergone Steering Committee review.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 6]
```diff
@@ -153,30 +146,32 @@ In `VideoStreamManager` add new `startRemoteDisplayStream`:
...
 
 ```java
 private List<VideoStreamingCapability> getSupportedCapabilities(
     Resolution minResolution,
     Resolution maxResolution,
-    Double constraintDiagonalMax,
-    AspectRatio ratioRange
+    Double minAspectRatio,
+    Double maxAspectRatio,
+    Double minScreenDiagonal
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:174]: These changes were not requested from the Steering Committee: https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-658826750.  We understand that they're incorrect, but will need a revision proposal to change, as the changes have not undergone Steering Committee review.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md (hunk 1 above)
Comment (dboltovskyi) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:118]: @theresalech Please find an update in a7fb1c588ea34c9d354abb58f57edbf26512c302

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md (hunk 5 above)
Comment (dboltovskyi) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md]: @theresalech Please notice these changes have been reverted in 04adb9c200ca4f7c69d20606870be148c0d3110a
We would like to go with this update through common SDLC process as you suggested.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md (hunk 6 above)
Comment (dboltovskyi) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md]: @theresalech Please notice these changes have been reverted in 04adb9c200ca4f7c69d20606870be148c0d3110a
We would like to go with this update through common SDLC process as you suggested.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 7]
```diff
@@ -118,10 +118,32 @@ While `diagonalScreenSize` is mandatory, `aspectRatio` and `resolution` are opti
 In `SDLStreamingMediaConfiguration`:
 
 ```objectivec
-@property (strong, nonatomic, nullable) VideoStreamingRange *supportedLandscapeStreamingRange;
-@property (strong, nonatomic, nullable) VideoStreamingRange *supportedPortraitStreamingRange;
+@property (strong, nonatomic, nullable) SDLVideoStreamingRange *supportedLandscapeStreamingRange;
+@property (strong, nonatomic, nullable) SDLVideoStreamingRange *supportedPortraitStreamingRange;
+@property (weak, nonatomic, nullable) id<SDLStreamingMediaDelegate> delegate;
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:123]:
```suggestion
@property (weak, nonatomic, nullable) id<SDLStreamingVideoDelegate> delegate;
```
See https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-642797352

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 8]
```diff
@@ -118,10 +118,32 @@ While `diagonalScreenSize` is mandatory, `aspectRatio` and `resolution` are opti
...
+```
+
+Where an object conforming to `SDLStreamingMediaDelegate` should implement the following methods:
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:126]:
```suggestion
Where an object conforming to `SDLStreamingVideoDelegate` should implement the following methods:
```

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 9]
```diff
@@ -118,10 +118,32 @@ While `diagonalScreenSize` is mandatory, `aspectRatio` and `resolution` are opti
...
+
+```objectivec
+@protocol SDLStreamingMediaDelegate <NSObject>
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:129]:
```suggestion
@protocol SDLStreamingVideoDelegate <NSObject>
```

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 10]
```diff
@@ -118,10 +118,32 @@ While `diagonalScreenSize` is mandatory, `aspectRatio` and `resolution` are opti
...
 
-@interface VideoStreamingRange : NSObject
+- (void)videoStreamingSizeDidUpdate:(CGSize)displaySize;
+- (void)videoStreamingSizeDoesNotMatch;
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md:132]:
```suggestion
```

This is not included in https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-642797352.  Please remove.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md [hunk 11]
```diff
@@ -118,10 +118,32 @@ While `diagonalScreenSize` is mandatory, `aspectRatio` and `resolution` are opti
...
+Whenever the video streaming capability changes `SDLStreamingVideoLifecycleManager` should call on the delegate:
+```objectivec
+[weakSelf.delegate videoStreamingSizeDidUpdate:displaySize];
+```
+In this case the client application will have a chance to update its view accordingly once it implements the delegate method.
+
+`SDLVideoStreamingRange` class should have the following properties:
+
+```objectivec
+@interface SDLVideoStreamingRange : NSObject
```
Comment (theresalech) [inline proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md]: I do not see this included in https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-642797352.  Please remove, and request as a new revision, or advise if it's in the comment and I missed it.  Thanks!

Comment (dboltovskyi) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md]: @theresalech This is just renaming of existing `SDLSupportedStreamingRange` to `SDLVideoStreamingRange` as suggested in https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-658826750 for item 3

Comment (yoooriii) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md]: Hello @theresalech, let me explain.
There was a question:
[3.2. How do we notify the app dev on changes?](https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-641424826)
and the [answer](https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-642797352)
The code above (lines 137-141) implements the requested functionality.

**Comment by dboltovskyi** [comment]
@theresalech We've made all the changes you suggested. Please advice if anything else needs to be updated?

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md (hunk 11 above)
Comment (theresalech) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md]: @yoooriii @dboltovskyi it doesn't appear this was part of what was agreed to by the Steering Committee: https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-658826750.  Please remove and include in your request for additional revisions.  Thank you!

**Comment by theresalech** [comment]
@dboltovskyi, thank you!  I've verified all changes and responded to the open comment.  Please let me know when this is ready for re-review.

File: proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md (hunk 11 above)
Comment (dboltovskyi) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md]: @theresalech 
Current update includes the following (as suggested in https://github.com/smartdevicelink/sdl_evolution/issues/983#issuecomment-658826750 for item 3):

1.
> For iOS, change streamingVideoDelegate to delegate.

2. 
> For Both iOS and Java Suite, use VideoStreamingRange

Please advise what part of the changes should be removed.

Comment (theresalech) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md]: The comment detailing Steering Committee vote includes the new class names, but not the logic as described in lines 126 - 140.  You are correct that `VideoStreamingRange` is included in the comment, so lines 143 - 146 can remain as is.  Thanks!

Comment (dboltovskyi) [reply proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md]: @theresalech Thank you for the clarification. Please find update in 32adadc

**Pull request merged**